presto-main/src/main/java/com/facebook/presto/server/HttpServerModule.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.server;
+
+import com.facebook.airlift.discovery.client.AnnouncementHttpServerInfo;
+import com.facebook.airlift.http.server.Authenticator;
+import com.facebook.airlift.http.server.HttpRequestEvent;
+import com.facebook.airlift.http.server.HttpServer;
+import com.facebook.airlift.http.server.HttpServerBinder;
+import com.facebook.airlift.http.server.HttpServerConfig;
+import com.facebook.airlift.http.server.HttpServerInfo;
+import com.facebook.airlift.http.server.HttpServerProvider;
+import com.facebook.airlift.http.server.LocalAnnouncementHttpServerInfo;
+import com.facebook.airlift.http.server.RequestStats;
+import com.facebook.airlift.http.server.TheAdminServlet;
+import com.facebook.airlift.http.server.TheServlet;
+import com.facebook.presto.server.security.AuthenticationFilter;
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Binder;
+import com.google.inject.Module;
+import com.google.inject.Provides;
+import com.google.inject.Scopes;
+
+import javax.servlet.Filter;
+import javax.servlet.Servlet;
+
+import java.util.List;
+import java.util.Set;
+
+import static com.facebook.airlift.configuration.ConfigBinder.configBinder;
+import static com.facebook.airlift.event.client.EventBinder.eventBinder;
+import static com.google.inject.multibindings.MapBinder.newMapBinder;
+import static com.google.inject.multibindings.Multibinder.newSetBinder;
+import static org.weakref.jmx.guice.ExportBinder.newExporter;
+
+public class HttpServerModule
```

Comment:
Curious. Why are we copying stuff over from `prestodb/airlift` 
https://github.com/prestodb/airlift/blob/master/http-server/src/main/java/com/facebook/airlift/http/server/HttpServerModule.java

Reply:
Global AuthenticationFilter(part of the airlift module) was bound in the HttpServerModule. We had to enhance this AuthenticationFilter to use the new InternalAuthenticationManager. So I copied over the HttpServerModule to presto and bind the new AuthenticationFilter here.

Reply:
I haven't had a chance to read through the entire code.
If binding in a new filter is the concern, can't we do something similar to what ranger module did for wiring in `RangerBasicAuthHttpRequestFilter`
https://github.com/prestodb/presto/blob/09d5b336d02952c81cd10538d0ebbf3fb53169e8/presto-hive/src/main/java/com/facebook/presto/hive/security/ranger/RangerBasedAccessControlModule.java#L36-L39

Reply:
I did something similar, added below line of code in ServerSecurityModule to bind the new filter, but it didn't take effect.
`newSetBinder(binder, Filter.class, TheServlet.class).addBinding()
                .to(AuthenticationFilter.class).in(Scopes.SINGLETON);`